src/io/frame.rs

```diff
@@ -366,7 +366,7 @@ impl<T: Io, C: Codec> Sink for Framed<T, C> {
             }
         }
 
-        self.codec.encode(item, &mut self.wr);
+        self.wr = self.codec.encode(item).unwrap();
```

Comment:
Here I believe we'll want to return the error to the caller rather than calling `unwrap`